kafka-client/src/main/java/dev/responsive/kafka/store/CommitBuffer.java

```diff
@@ -242,6 +287,78 @@ private AtomicWriteResult flush(final long offset, final UUID txnid) {
     );
     buffer.clear();
 
+    maybeTruncateChangelog(offset);
+    return AtomicWriteResult.success(partition);
+  }
+
+  private AtomicWriteResult drain(
+      final Collection<AtomicWriter<K>> writers,
+      final Function<AtomicWriter<K>, CompletionStage<AtomicWriteResult>> action
+  ) {
+    return drain(
+        writers,
+        action,
+        AtomicWriter::partition,
+        partition,
+        ResponsiveConfig.MAX_CONCURRENT_REMOTE_WRITES_DEFAULT
+    );
+  }
+
+  // Visible for Testing
```

Comment:
I seem to recall ksql often using the `@VisibleForTesting` annotation, WDYT about picking that up here? 

FWIW this would not fly in Kafka Streams because Bruno especially always tells everyone (correctly) to remove comments like this 😂 because they end up out of date/not being true more often than not 

(personally I don't really care)

Reply:
I didn't want to include Guava in our dependencies since we'd have to shade it to avoid our clients having version conflicts, otherwise I'd use the annotation